### ISSUE TITLE
parser: should report error on invalid left-hand side in assignment expression

### ISSUE THREAD

**Issue by kermanx** [oxc-project/oxc#9794]
According to the [standard](https://tc39.es/ecma262/#sec-assignment-operators-static-semantics-early-errors), the following code should have SyntaxError:

```js
({} &= a);
([] += b);
```

[Babel](https://ast-explorer.dev/#eNoliUsKgCAQQK8isyroBEK3aNe0UJMwTAfNIMy7N9HufSp4kLCrS2WTHJ0wAHHQSlvPbJi72sQzCtVjwsAtcqsYhEDIsSRjp5ssgmQ/4lo88/Bv8mVzIX9vXjA0aC9vCSK4), [acorn](https://ast-explorer.dev/#eNolyUEKAjEMRuGrlH+lMCcY8BbiKpsQsxjpNCXtCFJ7dyOze3xvIGPFi9/cxLfasaAGsJiXaIm+jJm+t8RXcvqbhQ0qKRFUdn6ot80KYQ3I3LV1wnL+ZoeL3j9Vz73b88jRVCbmDzZTJyA=) and [swc](https://ast-explorer.dev/#eNpdjEsKAjEMQK9SslKYEwy48B7ZhFqhw0xSkuCH2rsbdEBw93gPXocVZljoRpa1NocJWgi756AcdOgjvU6JjqjI4SRcR04JwZ7s9ECYg0ve9gPC9M2LfdqVViu7upQsSi5q/6VuTdTPZkW9Cv868oDxBt/zNfg=) reports this syntax error, while [TypeScript](https://ast-explorer.dev/#eNo9zLEKgCAQgOFXOW4qaGsq6Anam245SsKQU9SCMN89Q2j9fvgTGhzx4IvD6rWL2KErEG+nflgLNCnDMwG35EmK2WKJBIDQsOwn72pRPmgrhCMMQ1dbfcxato97koz5BUVjJJQ=) doesn't.

**Comment by kermanx** [comment]
My bad(
